File: IdeaProjects/Personal/Plugins/src/org/roadkill/plugins/intellij/CutLineAction.java
package org.roadkill.plugins.intellij;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.DataContext;
import com.intellij.openapi.editor.Editor;
import com.intellij.openapi.editor.actionSystem.EditorAction;
import com.intellij.openapi.editor.actionSystem.EditorWriteActionHandler;
import com.intellij.openapi.editor.actions.KillRingUtil;

/**
 * User: mike
 * Date: 12/28/11 3:36 PM
 */
public class CutLineAction extends EditorAction {

    public CutLineAction() {
        super(new Handler());
    }

    public static class Handler extends EditorWriteActionHandler {
        public void executeWriteAction(Editor editor, DataContext dataContext) {
            if (!editor.getSelectionModel().hasSelection()) {
                editor.getSelectionModel().selectLineAtCaret();
            }

            int selectionStart = editor.getSelectionModel().getSelectionStart();
            int selectionEnd = editor.getSelectionModel().getSelectionEnd();

            KillRingUtil.cut(editor, selectionStart, selectionEnd);
        }
    }

}

File: Roadkill/src/main/kotlin/org/roadkill/plugins/intellij/CutLineAction.kt
package org.roadkill.plugins.intellij

import com.intellij.openapi.actionSystem.DataContext
import com.intellij.openapi.editor.Caret
import com.intellij.openapi.editor.Editor
import com.intellij.openapi.editor.actionSystem.EditorAction
import com.intellij.openapi.editor.actionSystem.EditorWriteActionHandler
import com.intellij.openapi.editor.actions.KillRingUtil

class CutLineAction : EditorAction(Handler()) {

    class Handler : EditorWriteActionHandler() {
        override fun executeWriteAction(editor: Editor, caret: Caret?, dataContext: DataContext) {
            if (!editor.selectionModel.hasSelection()) {
                editor.selectionModel.selectLineAtCaret()
            }

            KillRingUtil.cut(
                editor, editor.selectionModel.selectionStart, editor.selectionModel.selectionEnd
            )
        }
    }

}